ompi/mca/osc/rdma/osc_rdma_component.c

```diff
@@ -625,6 +626,7 @@ static int allocate_state_shared (ompi_osc_rdma_module_t *module, void **base, s
     }
 
     do {
+        free(temp); // free from prior iteration. Should be initialized to NULL.
```

Comment:
Just move it outside the loop, we can safely reuse the buffer if there are multiple allgather.

Reply:
Fixed.